sdk/storage/azure-storage-blob-nio/src/main/java/com/azure/storage/blob/nio/AzureFileStore.java

```diff
@@ -0,0 +1,120 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.storage.blob.nio;
+
+import com.azure.storage.blob.BlobContainerClient;
+
+import java.io.IOException;
+import java.nio.file.FileStore;
+import java.nio.file.attribute.FileAttributeView;
+import java.nio.file.attribute.FileStoreAttributeView;
+import java.util.Objects;
+
+/**
+ * An {@code AzureFileStore} is backed by an Azure Blob Storage container.
+ * 
+ * {@inheritDoc}
+ */
+public final class AzureFileStore extends FileStore {
+    private final AzureFileSystem parentFileSystem;
+    private final BlobContainerClient containerClient;
+
+    AzureFileStore(AzureFileSystem parentFileSystem, String containerName) throws IOException {
+        // A FileStore should only ever be created by a FileSystem.
+        if (Objects.isNull(parentFileSystem)) {
+            throw new IllegalStateException("AzureFileStore cannot be instantiated without a parent FileSystem");
+        }
+        this.parentFileSystem = parentFileSystem;
+        this.containerClient = this.parentFileSystem.getBlobServiceClient().getBlobContainerClient(containerName);
+
+        try {
+            // This also serves as our connection check.
+            if (!this.containerClient.exists()) {
+                this.containerClient.create();
+            }
+        } catch (Exception e) {
+            throw new IOException("There was an error in establishing the existence of container: " + containerName, e);
+        }
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public String name() {
+        return this.containerClient.getBlobContainerName();
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public String type() {
+        return null;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public boolean isReadOnly() {
+        return false;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public long getTotalSpace() throws IOException {
+        return 0;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public long getUsableSpace() throws IOException {
+        return 0;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public long getUnallocatedSpace() throws IOException {
+        return 0;
+    }
```

Comment:
Is it better to return -1 to be extremely obvious that we don't care about these numbers instead of accidentally implying there's no space or something like that?

Reply:
Updated PR description to note this API is not implemented yet and does not need to be reviewed.